src/IssueViz/OpenInIde/OpenInIdeHandlerImplementation.cs

```diff
@@ -108,17 +109,40 @@ internal class OpenInIdeHandlerImplementation : IOpenInIdeHandlerImplementation
             visualization = visualizationProcessor.HandleConvertedIssue(visualization) ;
         }
         issueSelectionService.SelectedIssue = visualization;
-        
+
+        toolWindowService.Show(toolWindowId);
+
         var navigationResult = navigator.TryNavigatePartial(visualization);
-        HandleNavigationResult(navigationResult, toolWindowId, visualization);
+        if (navigationResult != NavigationResult.OpenedLocation)
+        {
+            HandleIncompleteNavigation(visualization, navigationResult);
+        }
+        else
+        {
+            logger.WriteLine(string.Format(OpenInIdeResources.DoneProcessingRequest, issueDetails?.Key));
+        }
+    }
+
+    private bool ValidateIssueNotNull<T>(T issueDetails, out string failureReason) where T : IOpenInIdeIssue
+    {
+        failureReason = default;
+
+        if (issueDetails is { Key: not null })
+        {
+            return true;
+        }
+
+        failureReason = OpenInIdeResources.ValidationReason_MalformedRequest;
+        return false;
```

Comment:
This is pretty hard to follow, the logic is somewhat jumping all over the place.
It should either do this, to clearly say what happens on happy path and what happens on invalid path
```
       if (issueDetails is { Key: not null })
       {
           failureReason = default;
            return true;
        }
        else
        {
            failureReason = OpenInIdeResources.ValidationReason_MalformedRequest;
            return false;
        }

```

or, make the failureReson coupled with the result value
```
      failureReason = issueDetails is { Key: not null } ? default : OpenInIdeResources.ValidationReason_MalformedRequest;
      return failureReason == default;
```

and that actually shows that the `bool` is redundant to the failureReason being null or not. So this should just return a string and the invocation should be something like
```
DetailsFailureReason(issueDetails) is { } failureReason
```

Reply:
> and that actually shows that the bool is redundant to the failureReason being null or not.

I did it in the TryGet way to be consistent with other validation conditions. While it's redundant, I think it reads better (especially since you can use `out var failureReason`)

Reply:
Changed the method structure, but did not change the signature